api/src/test/java/org/openmrs/EncounterTest.java

```diff
@@ -1406,75 +1425,31 @@ public void hasDiagnosis_shouldReturnFalseIfEncounterDoesNotHaveDiagnosis() {
 	 */
 	@Test
 	public void getAllConditions_shouldReturnAllConditions() {
-
-		Condition activeCondition = new Condition();
-		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
-		CodedOrFreeText freeText = new CodedOrFreeText();
-		freeText.setNonCoded("Free text");
-		activeCondition.setCondition(freeText);
-		encounter.addCondition(activeCondition);
-
-		Condition voidedCondition = new Condition();
-		voidedCondition.setVoided(true);
-		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
-		CodedOrFreeText freeText01 = new CodedOrFreeText();
-		freeText01.setNonCoded("Free text");
-		voidedCondition.setCondition(freeText01);
-		encounter.addCondition(voidedCondition);
-
-		assertEquals(2, encounter.getAllConditions(true).size());
+		
+		assertEquals(2, encounter.getConditions(true).size());
 	}
 
 	/**
 	 * @see Encounter#getConditions()
 	 */
 	@Test
 	public void getConditions_shouldReturnActiveConditions() {
-
-		Condition activeCondition = new Condition();
-		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
-		CodedOrFreeText freeText = new CodedOrFreeText();
-		freeText.setNonCoded("Free text");
-		activeCondition.setCondition(freeText);
-		encounter.addCondition(activeCondition);
-
-		Condition voidedCondition = new Condition();
-		voidedCondition.setVoided(true);
-		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
-		CodedOrFreeText freeText01 = new CodedOrFreeText();
-		freeText01.setNonCoded("Free text");
-		voidedCondition.setCondition(freeText01);
-		encounter.addCondition(voidedCondition);
-
+		
 		assertEquals(1, encounter.getConditions().size());
 	}
 
 	/**
-	 * @see Encounter#removeCondition()
+	 * @see Encounter#removeCondition(Condition)
 	 */
 	@Test
 	public void removeCondition_shouldRemoveConditions() {
 
-		Condition activeCondition = new Condition();
-		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
-		CodedOrFreeText freeText = new CodedOrFreeText();
-		freeText.setNonCoded("Free text");
-		activeCondition.setCondition(freeText);
-		encounter.addCondition(activeCondition);
 		// The condition should be voided
 		encounter.removeCondition(activeCondition);
 
-		Condition voidedCondition = new Condition();
-		voidedCondition.setVoided(true);
-		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
-		CodedOrFreeText freeText01 = new CodedOrFreeText();
-		freeText01.setNonCoded("Free text");
-		voidedCondition.setCondition(freeText01);
-		encounter.addCondition(voidedCondition);
-
 		Set<Condition> a = encounter.getConditions();
-		Set<Condition> b = encounter.getAllConditions(true);
+		Set<Condition> b = encounter.getConditions(true);
 		assertEquals(0, encounter.getConditions().size());
-		assertEquals(2, encounter.getAllConditions(true).size());
+		assertEquals(2, encounter.getConditions(true).size());
 	}
```

Comment:
What is this test actually testing?

Reply:
That the conditions are voided and not deleted.